File: backend/src/main/java/com/horain/chat/ChatResponse.java
package com.horain.chat;

import java.util.List;
import java.util.UUID;

/**
 * Response from the chat endpoint.
 */
public record ChatResponse(
        String assistantMessage,
        List<ToolCallRecord> toolCalls,
        Object data,
        UUID turnId
) {
}

File: backend/src/main/kotlin/com/horain/chat/ChatResponse.kt
package com.horain.chat

import java.util.UUID
import kotlin.jvm.JvmRecord

/**
 * Response from the chat endpoint.
 */
@JvmRecord
data class ChatResponse(
    val assistantMessage: String,
    val toolCalls: List<ToolCallRecord>,
    val data: Any?,
    val turnId: UUID?,
    /** Stable id for the chat thread; reuse on follow-up requests for LangSmith threads and DB grouping. */
    val conversationId: UUID
)
